x-pack/plugins/cases/common/types/api/configure/v1.ts

```diff
@@ -64,6 +68,36 @@ export const CustomFieldsConfigurationRt = limitedArraySchema({
   fieldName: 'customFields',
 });
 
+export const TemplateConfigurationRt = rt.strict({
+  /**
+   * key of template
+   */
+  key: regexStringRt({
+    codec: limitedStringSchema({ fieldName: 'key', min: 1, max: MAX_TEMPLATE_KEY_LENGTH }),
+    pattern: '^[a-z0-9_-]+$',
+    message: `Key must be lower case, a-z, 0-9, '_', and '-' are allowed`,
+  }),
+  /**
+   * name of template
+   */
+  name: limitedStringSchema({ fieldName: 'name', min: 1, max: MAX_TEMPLATE_NAME_LENGTH }),
+  /**
+   * description of templates
+   */
+  description: rt.string,
```

Comment:
Should this also be a `limitedStringSchema`?

Reply:
yeah, what should be the maximum length?

Reply:
added in https://github.com/elastic/kibana/pull/182575/commits/41deb1beeb129eb1849a1dad293de530190081cc